### ISSUE TITLE
Is it possible to pass environment overrides to Jinja?

### ISSUE THREAD

**Issue by rgitzelub** [cookiecutter/cookiecutter#972]
* Cookiecutter version: 1.5.1
* Template project url:
* Python version: 2.7.13
* Operating System:  OSX 10.12.5

### Description:

I am attempting to use Cookiecutter to generate some files for Ansible. Since they are both Jinja-based, and use the same delimiters, CC is trying to replace not just the {{cookiecutter:*}} placeholders, but the Ansible placeholders as well. :(

For example, something like this:
```
project: {{cookiecutter:projectName}}
subproject: database
stack_name: '{{project}}--{{stack_env}}--{{subproject}}'
```
CC fails trying to replace 'project'.

I found this reference as an example of how to configure Jinja to use something other than `{{}}` which would solve the problem:  https://groups.google.com/d/msg/ansible-project/EwEMbfJkQyg/doMDBFkWpEsJ

E.g. something like 
```
environment = jinja2.Environment(variable_start_string='@=', variable_end_string='=@')
```
would allow me to do this:
```
project: @=cookiecutter:projectName=@
subproject: database
stack_name: '{{project}}--{{stack_env}}--{{subproject}}'
```
But I have not been able to figure out how to configure CC to pass that configuration on to Jinja. Hopefully it's just a matter of my not reading the docs closely enough.

Thanks!

**Comment by luzfcb** [comment]
A simpler way to make cookiecutter not render jinja2 variables is to put what should not be processed inside `{% raw %}{% endraw %}` blocks

```
{% raw %}
 {{ ansible_jinja_variable }} 
{% endraw %}
```

a sample: https://github.com/HackSoftware/cookiecutter-django-ansible/blob/master/%7B%7B%20cookiecutter.ansible_project_slug%20%7D%7D/deploy.yml

**Comment by rgitzelub** [comment]
Ah!  I'd seen that approach for single variables, but hadn't thought to use it for blocks, as in the linked example.

I've _hundreds_ of Ansible placeholders, so wrapping each one would be a lot of work.  I'll take a look to see how inter-mixed the two kinds of placeholders are.

**Comment by rgitzelub** [comment]
There's also a PR that apparently fixes this: https://github.com/audreyr/cookiecutter/pull/946/files (based on https://github.com/audreyr/cookiecutter/issues/704)

**Comment by JoeAlamo** [comment]
If there are entire directories that you don't want cookiecutter to try to render, use https://cookiecutter.readthedocs.io/en/latest/advanced/copy_without_render.html

**Comment by insspb** [comment]
@rgitzelub Thank you for question. 
This question make a big discussion here. 

At final: 

* PR #1419 will allow to extend Jinja2 environment variables on template base. I.e. by template developer. As this is more required in some cases of template development. It and available in current master, and will be available in cookicutter 2.0
* Some cases can be solved with copy without render
* Some cases can be solved with hooks. 

All depends.
